clanker-deployer/DeployToken.ts

```diff
@@ -0,0 +1,646 @@
+  // For tokens with non-standard decimals (e.g., USDC with 6 decimals),
+  // you'll need to adjust the price calculation to account for the decimal difference:
+  // - If paired token has 6 decimals (like USDC): multiply desiredPrice by 10^12
+  // - General formula: desiredPrice * (10^(token1Decimals - token0Decimals))
+  // Where token0 is the token with the lower address (which could be either your token or paired token)
+  const logBase = 1.0001;
+  const tickSpacing = 200;
+  const rawTick = Math.log(desiredPrice) / Math.log(logBase);
+  const initialTick = Math.floor(rawTick / tickSpacing) * tickSpacing;
+
+  // Important: The contracts expect the tick to always be calculated as if the
+  // created token has a lower address than the paired token, even when its actual address is higher.
+  // Uniswap V3 sorts tokens by address, so this ensures consistent pricing.
+
+  // Generate salt
+  const { salt } = await generateSalt_v3_1(
+    deployerAddress,
+    formData.name,
+    formData.symbol,
+    formData.imageUrl || '',
+    JSON.stringify(clankerMetadata),
+    JSON.stringify(clankerSocialContext),
+    BigInt(parseUnits('100000000000', 18).toString()),
+    pairAddress,
+    BigInt(chainId)
+  );
+
+  // Format metadata and social context
+  const metadata = JSON.stringify(clankerMetadata);
+  const socialContext = JSON.stringify(clankerSocialContext);
+
+  // Calculate vesting duration
+  let vestingDuration = formData.vestingUnlockDate
+    ? getRelativeUnixTimestamp(Number(formData.vestingUnlockDate))
+    : BigInt(0);
+
+  // Fail early if vesting duration is too short (optional minimum check)
+  const MIN_VESTING_DURATION = BigInt(7 * 24 * 60 * 60); // 7 days in seconds
+  if (vestingDuration > 0 && vestingDuration < MIN_VESTING_DURATION) {
+    throw new Error(`Vesting duration is too short. Minimum duration is ${MIN_VESTING_DURATION} seconds.`);
+  }
+
+  // Helper to validate addresses
+  const validateAddress = (
+    address: string | undefined,
+    defaultAddress: `0x${string}`
+  ): `0x${string}` => {
+    if (!address || address.trim() === '' || !address.startsWith('0x')) {
+      return defaultAddress;
+    }
+    return address as `0x${string}`;
+  };
+
+  // Build the deployment config
+  const deploymentConfig = {
+    tokenConfig: {
+      name: formData.name,
+      symbol: formData.symbol,
+      salt: salt,
+      image: formData.imageUrl || '',
+      metadata: metadata,
+      context: socialContext,
+      originatingChainId: BigInt(chainId),
+    },
+    poolConfig: {
+      pairedToken: pairAddress,
+      tickIfToken0IsNewToken: initialTick || 0,
+    },
+    initialBuyConfig: {
+      // Select appropriate fee tier based on paired token
+      // 10000 = 1%, 3000 = 0.3%, 500 = 0.05%
+      // Some tokens like USDC might not have pools with 1% fee tier on certain chains
+      pairedTokenPoolFee: pairAddress === WETH_ADDRESS ? 10000 : 
```

Comment:
apologies for being annoying on this, can it be passed in top level with a default of 10000? this will break dev buys for tokens like native

Reply:
yep just hardcoded it to 10000